language-support/ts/daml-ledger/index.test.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+import type { EventEmitter } from 'events';
+import * as jtv from '@mojotech/json-type-validation';
+import { Template, ContractId } from '@daml/types';
+import Ledger, { Event, CreateEvent } from './index';
+
+type Foo = Record<string, unknown>
+
+const Foo = {
+  templateId: 'FooTemplateId',
+  decoder: () => jtv.object(),
+  keyDecoder: () => jtv.constant(undefined),
+} as unknown as Template<object>;
+
+
+type Message =
+  | { events: Event<Foo>[] }
+
+interface MockWebSocket {
+  serverOpen(): void;
+  serverSend(message: Message): void;
+  serverClose(event: {code: number; reason: string}): void;
+}
+
+let mockInstance = undefined as unknown as MockWebSocket;
+const mockConstructor = jest.fn();
+const mockSend = jest.fn();
+
+jest.mock('isomorphic-ws', () => class {
```

Comment:
Oh, so you can mock node modules inline after all. [The docs](https://jestjs.io/docs/en/manual-mocks) only seem to talk about the `__mocks__` way to do this.

Reply:
Yeah, that's what we do in `@daml/react` too: https://github.com/digital-asset/daml/blob/7185608f876372df776c5e74dd1e5274cb4e518e/language-support/ts/daml-react/index.test.ts#L21